src/Controls/src/Xaml/XamlParser.cs

```diff
@@ -368,7 +369,12 @@ static void GatherXmlnsDefinitionAttributes()
 				s_xmlnsDefinitions,
 				currentAssembly?.FullName,
 				(typeInfo) =>
-					Type.GetType($"{typeInfo.clrNamespace}.{typeInfo.typeName}, {typeInfo.assemblyName}"));
+				{
+					var t = Type.GetType($"{typeInfo.clrNamespace}.{typeInfo.typeName}, {typeInfo.assemblyName}");
+					if (t is not null && t.IsPublicOrVisibleInternal(currentAssembly))
+						return t;
+					return null;
```

Comment:
This is the real fix for the XAML parser. Instead of just selecting the matching type, make sure it is a visible one. The `GetTypeReference` extension method loops over all combinations of assemblies/types/suffixes and some of those are valid, but inappropriate.